src/GroupManager.php

```diff
@@ -93,6 +101,10 @@ public function addGroup($entity_type_id, $bundle_id) {
     $editable->set('groups', $groups);
     $saved = $editable->save();
 
+    // Notify other module we added a new group.
+    // todo: should this be an event?
+    $this->moduleHandler->invokeAll('og_group_created', [$entity_type_id, $bundle_id]);
```

Comment:
I am of a different opinion, I prefer events :)

0. Every listener is contained in a separate class. This makes code reuse across projects easier. You can just copy the file into a new project without having to extract it from a module file.
0. It's easier to test. Since it's OO code and uses dependency injection you can use unit tests to test it and can mock all dependencies. This is much harder to do with hooks, in practice you would usually want to enable the module, meaning you need slower kernel tests to test it. You will then also inherit all the other procedural code which might not strictly be needed for the thing you are testing.
0. The listeners get lazy loaded. Hook implementations usually reside in a module file alongside dozens of hundreds of other hook implementations which might or might not be needed in the current request.
0. This is called from OO code, I personally find it cleaner to stay in 'OO-space' rather than delegating to procedural code if there is no compelling reason to do so.